### ISSUE TITLE
[compile] meta-llama correctness issue

### ISSUE THREAD

**Issue by angelayi** [pytorch/pytorch#161457]
### 🐛 Describe the bug

There seems to be an accuracy issue in torch.compile when running meta-llama/Llama-3.2-1B with bfloat16.

Things I've tried: 
* torch.compile backend eager/aot_eager/aot_eager_decomp_partition all work fine, so I guess this is an inductor issue 
* dtype=float32 works fine too 
* TORCHINDUCTOR_EMULATE_PRECISION_CASTS=1 does not seem to fix the issue

repro script:
```python
    def test_meta_llama(self):
        from transformers import AutoModelForCausalLM, AutoTokenizer
        import torch
        import time 
        import copy

        model_name = "meta-llama/Llama-3.2-1B"
        tokenizer = AutoTokenizer.from_pretrained(model_name)
        model = AutoModelForCausalLM.from_pretrained(
            model_name, torch_dtype=torch.bfloat16, device_map="cuda"
        )
        model.eval()
        
        model.generation_config.do_sample = False
        model.generation_config.use_cache = True
        model.generation_config.cache_implementation = "static"
        model.generation_config.max_new_tokens = 2000
        model.generation_config.pad_token_id = tokenizer.eos_token_id
        model.generation_config.temperature = 0.0

        vocab_size = tokenizer.vocab_size
        input_ids = torch.randint(
            low=0,
            high=vocab_size,
            size=(1, 1000),
            device="cuda",
            dtype=torch.long,
        )
        example_inputs = {"input_ids": input_ids}

        model_copy = copy.deepcopy(model)

        start = time.time()
        res1 = model_copy(**example_inputs).logits
        print(time.time() - start)

        model_copy = copy.deepcopy(model)
        model_copy.forward = torch.compile(model_copy.forward)

        start = time.time()
        res2 = model_copy(**example_inputs).logits
        print(time.time() - start)
        
        self.assertTrue(torch.allclose(res1, res2, atol=0.001))
```

### Versions

main, transformers 4.52.4

cc @ezyang @gchanan @zou3519 @kadeng @msaroufim @chauhang @penguinwu @voznesenskym @EikanWang @jgong5 @Guobing-Chen @XiaobingSuper @zhuhaozhe @blzheng @wenzhe-nrv @jiayisunx @ipiszy @chenyang78 @muchulee8 @amjames @aakhundov @coconutruben

**Comment by BoyuanFeng** [comment]
related issue: #159855

**Comment by anijain2305** [comment]
I think the accuracy issue still exists but just FYI - if you do - `model.generation_config.cache_implementation = "static"`, compile is enabled by default.

**Comment by zou3519** [comment]
We should compare to a float64 baseline to see how far away (compile might be closer to float64 baseline than eager, it happens sometimes).
